harness/src/network.rs

```diff
@@ -172,6 +178,12 @@ impl Network {
         }
     }
 
+    /// Call `self.read_messages()` and then send them.
```

Comment:
Seems this doesn't filter either?

Reply:
I also changed `read_message`, now it does filter.